fd/diff.go

```diff
@@ -19,25 +18,30 @@ type Point struct {
 	Coeff float64
 }
 
-// Method represents a finite difference formula that approximates
-// the derivative of order k of a function f at x as
-//  d^k f(x) ≅ (1 / h^k) * \sum_i Coeff_i * f(x + h * Loc_i),
-// where h is a small positive step.
-type Method struct {
+// Formula represents a finite difference formula on a regularly spaced grid
+// that approximates the derivative of order k of a function f at x as
+//  d^k f(x) ≅ (1 / Step^k) * \sum_i Coeff_i * f(x + Step * Loc_i).
+type Formula struct {
 	// Stencil is the set of sampling Points which are used to estimate the
 	// derivative. The locations will be scaled by Step and are relative to x.
 	Stencil []Point
 	Order   int     // The order of the approximated derivative.
-	Step    float64 // Default step size for the method.
+	Step    float64 // Default step size for the formula.
```

Comment:
Ehhh, that's a pretty small step size.

The majority of my usage for Derivative / Gradient is estimating the derivative for optimization purposes. This is in alignment with what other people are using that I talk with. Most of those kinds of functions have sufficient numerical noise that you want the step to be somewhere in the 10-8 to 10-4 range. I don't quite remember my initial motivation for it being part of the Formula rather than default ot 1e-6, for example.